### PR TITLE
tests: lib: cmsis_dsp: move svm to new ztest API

### PR DESCRIPTION
**Pull request by 1xlibiao**
Move test svm to use new ztest API.

Signed-off-by: li biao <biao1x.li@intel.com>

### CODE REVIEW

File: tests/lib/cmsis_dsp/svm/src/f16.c [hunk 1]
```diff
@@ -41,8 +41,8 @@
 		params[svec_count + (vec_dims * svec_count) + 1]; \
 	const float16_t gamma = \
 		params[svec_count + (vec_dims * svec_count) + 2]
-
-static void test_arm_svm_linear_predict_f16(void)
+#ifdef CONFIG_CMSIS_DSP_FLOAT16
```
Comment (stephanosio) [inline tests/lib/cmsis_dsp/svm/src/f16.c:44]: This should not be necessary.

`f16.c` is only compiled if `CONFIG_CMSIS_DSP_FLOAT16=y` in the first place.

**Review by stephanosio — CHANGES_REQUESTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/47918#discussion_r922955107

File: tests/lib/cmsis_dsp/svm/src/f16.c (hunk 1 above)
Comment (1xlibiao) [reply tests/lib/cmsis_dsp/svm/src/f16.c]: > This should not be necessary.
> 
> `f16.c` is only compiled if `CONFIG_CMSIS_DSP_FLOAT16=y` in the first place.

ok,I have changed it,thanks

**Review by stephanosio — APPROVED** [review]

**Review by stephanosio — DISMISSED** [review]

**Comment by enjiamai** [comment]
Hi @1xlibiao , please rebase this PR, thanks.

**Comment by 1xlibiao** [comment]
> Hi @1xlibiao , please rebase this PR, thanks.

ok

**Review by chen-png — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Pull request merged**